python/lsst/ip/isr/isrMock.py

```diff
@@ -0,0 +1,915 @@
+        dtype=bool,
+        default=True,
+        doc="Add dark signal to data.",
+    )
+    doFlat = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Add flat signal to data.",
+    )
+    doFringe = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Add fringe signal to data.",
+    )
+
+    # Datasets to create and return instead of generating an image.
+    doTransmissionCurve = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Return a simulated transmission curve.",
+    )
+    doDefects = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Return a simulated defect list.",
+    )
+    doBrighterFatter = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Return a simulated Brighter-Fatter kernel.",
+    )
+    doCrosstalkCoeffs = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Return the matrix of crosstalk coefficients.",
+    )
+    doDataRef = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Return a simulated gen2 butler dataRef.",
+    )
+    doGenerateImage = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Generate an output image?",
+    )
+    doGenerateData = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Generate some other data structure?",
+    )
+    doGenerateDict = pexConfig.Field(
+        dtype=bool,
+        default=False,
+        doc="Generate a dict of exposure amplifiers instead of an afwImage.Exposure.",
+    )
+
+
+class IsrMock(pipeBase.Task):
+    r"""Class to generate consistent mock images for ISR testing.
+
+    ISR testing currently relies on one-off fake images that do not
+    accurately mimic the full set of detector effects.  This class
+    uses the test camera/detector/amplifier structure defined in
+    `lsst.afw.cameraGeom.testUtils` to avoid making the test data
+    dependent on any of the actual obs package formats.
+    """
+    ConfigClass = IsrMockConfig
+    _DefaultName = "isrMock"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.rng = np.random.RandomState(self.config.rngSeed)
+        self.crosstalkCoeffs = np.array([[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                         [1e-2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                         [1e-2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                         [1e-2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                         [1e-2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                         [1e-2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                         [1e-2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                         [1e-2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]])
+
+        self.bfKernel = np.array([[1., 4., 7., 4., 1.],
```

Comment:
I haven't dealt with BF kernels enough to have a good handle on their shapes.  I wanted something flux conserving, as well as something small enough to not require loading from an external source.  This kernel does run through the ISR brighterFatterCorrection code, which was my other main constraint.

Reply:
If it runs through and doesn't totally ruin everything then that's probably fine. Otherwise I can dig you out a more realistic one, they're not huge, usually 11x11 or 19x19 at worst I think.